bot/exts/filters/filtering.py

```diff
@@ -291,6 +304,71 @@ async def filter_eval(self, result: str, msg: Message) -> bool:
 
         return filter_triggered
 
+    async def _process_autoban(self, reason: str, message: discord.Message) -> None:
+        """Check if a filter triggers an autoban, and process it if so."""
+        # If the filter reason contains `[autoban]`, we want to auto-ban the user
+        if reason and "[autoban]" in reason.lower():
+            # Create a new context, with the author as is the bot, and the channel as #mod-alerts.
+            # This sends the ban confirmation directly under watchlist trigger embed, to inform
+            # mods that the user was auto-banned for the message.
+            context = await self.bot.get_context(message)
+            context.guild = self.bot.get_guild(Guild.id)
+            context.author = context.guild.get_member(self.bot.user.id)
+            context.channel = self.bot.get_channel(Channels.mod_alerts)
+            context.command = self.bot.get_command("tempban")
+
+            await context.invoke(
+                context.command,
+                message.author,
+                arrow.utcnow() + AUTO_BAN_DURATION,
+                reason=AUTO_BAN_REASON
+            )
+
+    async def _process_redirect(self, url: str, message: discord.Message, stats: Stats) -> None:
+        """Unfurl redirect, run the URL filters on it, and check if further punishment is required."""
+        log.trace(f"Processing redirects for {message.id} by {message.author}.")
+        filter_data = "filter_redirects", self.filters["filter_redirects"]
+        result = await services.unfurl_url(url, max_per_attempt=3, max_continues=0, use_cache=True)
+        reason = None
+
+        if result is None:
+            log.debug("Failed to unfurl a redirect.")
+            message_extend = ". URL could not be unfurled for an unknown reason."
+            ping = True
+
+        elif result.error is None:
+            has_urls = await self._has_urls(result.destination)
+            if has_urls[0]:
+                log.debug("Unfurled redirect, and destination was blacklisted.")
+                message_extend = (
+                    ", and tripped a domain filter.\n\n"
+                    f"Destination: `{result.destination}`\n"
+                    f"Redirects: {result.depth}\n\n"
+                    "Tripped "
+                )
+                if has_urls[1] is not None:
+                    reason = has_urls[1]
+                    await self._process_autoban(reason, message)
+                ping = True
+
+            else:
+                log.debug("Unfurled redirect, but destination was not blacklisted.")
+                message_extend = ". URL unfurled, but destination was not blacklisted. Tripped "
+                ping = False
```

Comment:
Any blacklisted redirect domains will send an alert no matter what. The only thing that changes is the mod ping, because I don't think we need to alert whenever someone uses a link shortener.

The reason the link isn't there is to avoid making the embed larger than needed, and let mods run the unfurl command if they want that info. If we want to add it, it'll be pretty easy.

Reply:
We could run it by the mods regarding the ping, but I do think there's no reason to not show the unfurl information when it's already fetched.